pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java

```diff
@@ -100,6 +105,27 @@ public boolean expireMessages(int messageTTLInSeconds) {
         }
     }
 
+    private boolean checkExpiryByLedgerClosureTime(ManagedCursor cursor, int messageTTLInSeconds) {
+        if (cursor instanceof ManagedCursorImpl managedCursor) {
+            ManagedLedgerImpl managedLedger = (ManagedLedgerImpl) managedCursor.getManagedLedger();
+            Position deletedPosition = managedCursor.getMarkDeletedPosition();
+            SortedMap<Long, MLDataFormats.ManagedLedgerInfo.LedgerInfo> ledgerInfoSortedMap =
+                    managedLedger.getLedgersInfo()
+                            .subMap(deletedPosition.getLedgerId(), managedLedger.getLedgersInfo().lastKey());
+            long expiryLedgerId = -1;
+            for (MLDataFormats.ManagedLedgerInfo.LedgerInfo ledgerInfo : ledgerInfoSortedMap.values()) {
+                if (!MessageImpl.isEntryExpired(messageTTLInSeconds, ledgerInfo.getTimestamp())) {
+                    break;
+                }
+                expiryLedgerId = ledgerInfo.getLedgerId();
```

Comment:
```suggestion
                if (!ledgerInfo.hasTimestamp()) {
                    break;
                }
                if (!MessageImpl.isEntryExpired(messageTTLInSeconds, ledgerInfo.getTimestamp())) {
                    break;
                }
                expiryLedgerId = ledgerInfo.getLedgerId();
```

Is it better to add a check to make sure the ledger has timestamp to avoid a potential NPE?
I know it shouldn't happen because the loop will not touch the last ledger.